crates/apollo-router-core/src/response.rs

```diff
@@ -93,7 +93,7 @@ impl Response {
                 })?;
 
         for node in nodes {
-            node.deep_merge(&value);
+            node.deep_merge(value);
```

Comment:
we might want deep_merge to accept a `Value` instead of `&Value` here. Since it's merging two `Value` objects, it could take nodes from the argument without cloning them (this is what we're doing right now: https://github.com/apollographql/router/blob/main/crates/apollo-router-core/src/json_ext.rs#L132 )

Reply:
It currently takes a reference and seems to work fine. If we decided that deep_merge() needs an owned Value, we could re-work things at that time to address that requirement.